CS 410 Assignment 3/src/ass4/ControllerNotePad.java

```diff
@@ -0,0 +1,222 @@
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JMenu;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+import javax.swing.JTextPane;
+import javax.swing.text.Position;
+import javax.swing.text.StyledDocument;
+
+public class ControllerNotePad extends JFrame implements ActionListener{
+	ModelNotePad model=new ModelNotePad();
+	
+	
+    public ControllerNotePad() {
+    	 
+        setTitle("A Simple Notepad Tool");
+        model.getFileMenu().add(model.getOpen());
+        model.getFileMenu().addSeparator();
+        model.getFileMenu().add(model.getNewFile());
+        model.getFileMenu().addSeparator();
+        model.getFileMenu().add(model.getSaveFile());     
+        model.getFileMenu().addSeparator();
+        model.getFileMenu().add(model.getPrintFile());
+        model.getFileMenu().addSeparator();
+        
+        model.getFileMenu().add(model.getRecent());
+        
+        model.getEditMenu().add(model.getReplace()); 
+        model.getEditMenu().add(model.getUndo());
+         model.getEditMenu().add(model.getCopy());
+         model.getEditMenu().add(model.getPaste());
+         
+        model.getNewFile().addActionListener(this);
+        
+        model.getNewFile().setActionCommand("new");
+        model.getSaveFile().addActionListener(this);
+        
+        model.getSaveFile().setActionCommand("save");
+        model.getPrintFile().addActionListener(this);
+        
+        model.getPrintFile().setActionCommand("print");
+        
+        model.getCopy().addActionListener(this);
+        model.getCopy().setActionCommand("copy");
+        model.getPaste().addActionListener(this);
+        model.getPaste().setActionCommand("paste");
+        model.getUndo().addActionListener(this);
+        model.getUndo().setActionCommand("undo");
+        model.getOpen().addActionListener(this);
+        model.getOpen().setActionCommand("open");
+        
+        model.getReplace().addActionListener(this);
+        model.getReplace().setActionCommand("replace");
+        
+        model.getRecent().setMnemonic(KeyEvent.VK_S);
+
+        
+     //   recent.addMenuListener(new MenuListener());
+
+        model.getRecent().addActionListener(this);
+        model.getRecent().setActionCommand("recent");
+        
+        model.getMenuBar().add(model.getFileMenu());
+        model.getMenuBar().add( model.getEditMenu());
+        setJMenuBar(model.getMenuBar());
+        add(new JScrollPane(model.getTextArea()));
+        setPreferredSize(new Dimension(600,600));
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setVisible(true);
+        pack();
+    
+    }
+    @Override
+    public void actionPerformed(ActionEvent e) {

```

Comment:
I would recommend taking out this if-else statement by using lambda functions to add ActionListeners. That way you won't have a lengthy if-else statement. An if-else statement can be a code smell because it makes it harder to add new functionality, and it is not very readable.